Sources/tart/Fetcher.swift

```diff
@@ -2,45 +2,35 @@ import Foundation
 import AsyncAlgorithms
 
 class Fetcher: NSObject, URLSessionTaskDelegate, URLSessionDelegate, URLSessionDataDelegate {
-    let responseCh = AsyncThrowingChannel<URLResponse, Error>()
+  func fetch(_ request: URLRequest) async throws -> (AsyncThrowingChannel<Data, Error>, HTTPURLResponse) {
     let dataCh = AsyncThrowingChannel<Data, Error>()
 
-    func fetch(_ request: URLRequest) async throws -> (AsyncThrowingChannel<Data, Error>, URLResponse) {
-        let task = URLSession.shared.dataTask(with: request)
-        task.delegate = self
-        task.resume()
+    let (data, response) = try await URLSession.shared.data(for: request)
```

Comment:
I wonder if that needs to be inside the `Task` 🤔  But probably it's better to keep it here for better error handling and stacktrace understanding.

Reply:
Since we're returning `URLResponse` immediately, it makes sense to only process `Data` inside of a `Task`.